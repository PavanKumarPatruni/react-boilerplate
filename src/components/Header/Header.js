import React from 'react';
import PropTypes from 'prop-types';

import './Header.scss';

const propTypes = {
  title: PropTypes.string,
};

const Header = ({ title }) => {
  return (
    <div>
      <h1>{title}</h1>
    </div>
  );
};

Header.propTypes = propTypes;

Header.defaultProps = {
  title: 'React Setup',
};

export default Header;
