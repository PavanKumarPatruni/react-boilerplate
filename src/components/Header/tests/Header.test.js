import React from 'react';
import { shallow, mount } from 'enzyme';

import Header from '../Header';

describe('Header component', () => {
  const props = {
    title: 'React',
  };

  test('match snapshot correctly', () => {
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly', () => {
    props.Header = '';
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly', () => {
    const wrapper = mount(<Header {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
