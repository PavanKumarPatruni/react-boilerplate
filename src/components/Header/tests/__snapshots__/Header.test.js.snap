// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component match snapshot correctly 1`] = `
<div>
  <h1>
    React
  </h1>
</div>
`;

exports[`Header component match snapshot correctly 2`] = `
<div>
  <h1>
    React
  </h1>
</div>
`;

exports[`Header component match snapshot correctly 3`] = `
<Header
  Header=""
  title="React"
>
  <div>
    <h1>
      React
    </h1>
  </div>
</Header>
`;
